### PR TITLE
Rule status vs time

### PR DESCRIPTION
**Pull request by FernandoGarzon**

### CODE REVIEW

File: cms/check_rule_counts [hunk 1]
```diff
@@ -148,44 +152,117 @@ if __name__ == '__main__':
...
+
+            # Number of replicating rules older than x
+
+            query = (session.query(func.count(models.ReplicationRule.id))
+                     .filter(models.ReplicationRule.state == RuleState.REPLICATING)
+                     .filter(models.ReplicationRule.created_at <= timeLimit))
+            result = query.scalar() or 0
+            print('rules.no_of_rules.replicating.%s.replicating_rules_cnt %s' % (a_name, result))
+
+            prom_labels = {'age': a_name.split('_')[1]}
```
Comment (ericvaandering) [inline cms/check_rule_counts:232]: This seems to be a repeat (with no change in intermediate value of the code on 217 and 202 (a_name and extra_prom_labels are not changing)

File: cms/check_rule_counts [hunk 2]
```diff
@@ -148,44 +152,117 @@ if __name__ == '__main__':
...
 
             query = (session.query(func.sum(models.ReplicationRule.locks_stuck_cnt))
                      .filter(models.ReplicationRule.state == RuleState.STUCK)
                      .filter(models.ReplicationRule.created_at <= timeLimit))
             result = query.scalar() or 0
             print('rules.no_of_files.stuck.%s.sum_locks_stuck_cnt %s' % (a_name, result))
 
-            prom_labels = {'Stuck_cnt': 'Stuck_files.%s.Sum' % a_name}
+            prom_labels = {'age': a_name.split('_')[1]}
             prom_labels.update(extra_prom_labels)
```
Comment (ericvaandering) [inline cms/check_rule_counts:248]: Same

File: cms/check_rule_counts [hunk 3]
```diff
@@ -148,44 +152,117 @@ if __name__ == '__main__':
...
+            # Number of Replicating files
+
             query = (session.query(func.sum(models.ReplicationRule.locks_replicating_cnt))
                      .filter(models.ReplicationRule.state.in_([RuleState.STUCK, RuleState.REPLICATING]))
                      .filter(models.ReplicationRule.created_at <= timeLimit))
             result = query.scalar() or 0
             print('rules.no_of_files.replicating.%s.sum_locks_replicating_cnt %s' % (a_name, result))
 
-            prom_labels = {'Replicating_cnt': 'Replicating_files.%s.Sum' % a_name}
+            prom_labels = {'age': a_name.split('_')[1]}
```
Comment (ericvaandering) [inline cms/check_rule_counts:262]: Same

File: cms/check_rule_counts [hunk 4]
```diff
@@ -200,5 +277,4 @@ if __name__ == '__main__':
         sys.exit(UNKNOWN)
     finally:
         session.remove()
-    sys.exit(OK)
-
+    sys.exit(OK)
```
Comment (ericvaandering) [inline cms/check_rule_counts:280]: Please have a carriage return at the end of the file

**Review by ericvaandering — CHANGES_REQUESTED** [review]

**Comment by FernandoGarzon** [comment]
Updated. Unnecessary dictionaries removed.

**Comment by FernandoGarzon** [comment]
Counts of state of rules and locks by activity added. No further changes are intended to be added to this probe.

**Pull request merged**